Extension/c_cpp_properties.schema.json

```diff
@@ -70,9 +70,22 @@
                         ]
                     },
                     "compileCommands": {
-                        "markdownDescription": "Full path to `compile_commands.json` file for the workspace.",
-                        "descriptionHint": "Markdown text between `` should not be translated or localized (they represent literal text) and the capitalization, spacing, and punctuation (including the ``) should not be altered.",
-                        "type": "string"
+                        "oneOf": [
+                            {
+                                "type": "string",
+                                "default": ""
+                            },
+                            {
+                                "type": "array",
+                                "items": {
+                                    "type": "string"
+                                },
+                                "uniqueItems": true,
+                                "default": []
+                            }
+                        ],
```

Comment:
Right now, if "C_Cpp.default.compileCommands" is added in the settings.json, VS Code initializes it to the invalid value of `null`. It seems like we should use

```json
                        "default": [
                            ""
                        ],
```

Or maybe just `[]`.

I don't the "default" inside of the "oneOf" is used.

Also, it seems like this code below in `getAsArrayOfStringsOrUndefined`
```ts
        if (setting.default !== undefined) {
            console.error(`Default value for ${settingName} is expected to be undefined.`);
        }
```
should be removed if that default is added.

Reply:
Please do not remove that check in `getAsArrayOfStringsOrUndefined`. Its purpose is to catch incorrect settings validation. I'll take a look at the PR and recommend a different approach.

Reply:
I pushed a commit that addresses my concerns and also fixes a bug I noticed with the settings tracker not noticing when the value changes.  Please review and let me know if there are any issues.

Reply:
ok, seems good to me